packages/react/src/components/DetailsList/DetailsHeader.base.tsx

```diff
@@ -425,6 +425,21 @@ export class DetailsHeaderBase
     this._draggedColumnIndex = -1;
   };
 
+  private _computeColumnIndexOffset = (showCheckbox: boolean) => {
+    const hasGroupExpander = this.props.groupNestingDepth! > 0;
```

Comment:
Avoid non-null assertions `!`. Could this instead be a truthy check? 
```suggestion
    const hasGroupExpander = !!this.props.groupNestingDepth;
```

Or if checking for non-negative values is important:
```suggestion
    const hasGroupExpander = !!this.props.groupNestingDepth && this.props.groupNestingDepth > 0;
```